core/internal/api/api.go

```diff
@@ -121,7 +129,11 @@ type ClientOptions struct {
 	// starts a new timeout.
 	NonRetryTimeout time.Duration
 
-	// Additional headers to pass in each request.
+	// Additional headers to pass in each request to the backend.
+	//
+	// Note that these are only passed when communicating with the W&B backend.
+	// In particular, they are not sent if using this client to send
+	// arbitrary HTTP requests.
```

Comment:
we sometimes get extra headers, when doing file upload. see here: sendFile, but this is going to use the non wandb endpoint

Reply:
https://github.com/wandb/wandb/blob/main/core/pkg/server/sender.go#L998-L1007

Reply:
this is also a guardrail for if the client ever makes unexpected non-W&B requests